src/cascadia/TerminalSettingsEditor/SettingContainer.cpp

```diff
@@ -172,12 +160,113 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+                {
+                    const std::wstring templateText{ RS_(L"SettingContainer_OverrideIntro") };
+                    const auto pos{ templateText.find(L"{}") };
+                    if (pos != std::wstring::npos)
+                    {
+                        // Append Message Prefix
+                        {
+                            Documents::Run msgPrefix{};
+                            msgPrefix.Text(templateText.substr(0, pos));
+                            tb.Inlines().Append(msgPrefix);
+                        }
+
+                        // Append Message Target
+                        {
+                            Documents::Run msgTarget{};
+                            const auto& settingSrc{ SettingOverrideSource() };
+                            if (!settingSrc)
+                            {
+                                // no source; provide system-default view
+                                msgTarget.Text(RS_(L"SettingContainer_OverrideTargetSystemDefaults"));
+                                tb.Inlines().Append(msgTarget);
+                            }
+                            else if (const auto& profile{ settingSrc.try_as<Model::Profile>() })
+                            {
+                                const auto originTag{ profile.Origin() };
+                                if (originTag == Model::OriginTag::InBox)
+                                {
+                                    // in-box profile
+                                    msgTarget.Text(profile.Name());
+                                    tb.Inlines().Append(msgTarget);
+                                }
+                                else if (originTag == Model::OriginTag::Generated)
+                                {
+                                    // from a dynamic profile generator
+                                    const auto& profileSource{ profile.Source() };
+                                    if (profileSource == WslGeneratorNamespace)
+                                    {
+                                        msgTarget.Text(RS_(L"SettingContainer_OverrideTargetWSLGenerator"));
+                                    }
+                                    else if (profileSource == PowershellCoreGeneratorNamespace)
+                                    {
+                                        msgTarget.Text(RS_(L"SettingContainer_OverrideTargetPowershellGenerator"));
+                                    }
+                                    else if (profileSource == AzureGeneratorNamespace)
+                                    {
+                                        msgTarget.Text(RS_(L"SettingContainer_OverrideTargetAzureCloudShellGenerator"));
+                                    }
+                                    else
+                                    {
+                                        // TODO #1690: add special handling for proto-extensions here

```

Comment:
Chatted with @PankajBhojwani about proto-extensions. This should mostly just work. Only changes that would be needed for proto-extensions are...
- tag profiles from proto-extensions
- use `Profile::Source` to create msgTarget
- add a hyperlink for a read-only view of the proto-extension